knoten/sensor_utils.py

```diff
@@ -0,0 +1,296 @@
+from knoten import csm, utils, bundle
+from csmapi import csmapi
+import numpy as np
+
+def phase_angle(image_pt, sensor):
+    """
+    Computes and returns phase angle, in degrees for a given image point.
+   
+    Phase Angle: The angle between the vector from the intersection point to
+    the observer (usually the spacecraft) and the vector from the intersection
+    point to the illuminator (usually the sun).
+
+    Parameters
+    ----------
+    image_pt : tuple
+               Pair of x, y (sample, line) coordinates in pixel space
+
+    sensor : object
+             A CSM compliant sensor model object
+
+    Returns
+    -------
+     : np.ndarray
+            phase angle in degrees
+    """
+    if not isinstance(image_pt, csmapi.ImageCoord):
+        image_pt = csmapi.ImageCoord(*image_pt)
+
+    sensor_state = csm.get_state(sensor, image_pt)
+    ground_pt = csm.generate_ground_point(0.0, image_pt, sensor)
+  
+    sunEcefVec = sensor.getIlluminationDirection(ground_pt)
+    illum_pos = utils.Point(ground_pt.x - sunEcefVec.x, ground_pt.y - sunEcefVec.y, ground_pt.z - sunEcefVec.z)
+
+    vec_a = utils.Point(sensor_state["sensorPos"].x - ground_pt.x, 
+                        sensor_state["sensorPos"].y - ground_pt.y, 
+                        sensor_state["sensorPos"].z - ground_pt.z)
+
+    vec_b = utils.Point(illum_pos.x - ground_pt.x, 
+                        illum_pos.y - ground_pt.y, 
+                        illum_pos.z - ground_pt.z)
+  
+    return np.rad2deg(utils.sep_angle(vec_a, vec_b))
```

Comment:
I was trying to match the output of campt which convert these to degrees. Wasn't sure on the use case whether it should be in radians or degrees. I can switch them back to radians.

Reply:
Sorry, didn't mean to imply that it needed to happen. Was just curious why the change from `rad2deg`. I don't have a reason it should be in rad and if we need it in deg we can always apply the conversion as needed. This can stay